staging/src/k8s.io/apiserver/pkg/authentication/token/cache/cached_token_authenticator.go

```diff
@@ -98,5 +110,17 @@ func (a *cachedTokenAuthenticator) AuthenticateToken(ctx context.Context, token
 }
 
 func keyFunc(auds []string, token string) string {
```

Comment:
A benchmark for this might be useful. I'm more worried about allocations than CPU.

Reply:
I'm also worried that `"%#v"` might print out some pointers instead of the contents; can you write a test proving that the hash value for:
* different aud pointers but same contents returns the same value?
* and: same aud pointers but different values (e.g., due to pointer reuse) returns a different value?